apis/database/v1alpha1/redshift_types.go

```diff
@@ -0,0 +1,713 @@
+	// through an Internet gateway. For more information about provisioning clusters
+	// in EC2-VPC, go to Supported Platforms to Launch Your Cluster (https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html#cluster-platforms)
+	// in the Amazon Redshift Cluster Management Guide.
+	// +optional
+	ElasticIP *string `json:"elasticIP,omitempty"`
+
+	// Encrypted defines whether your data in the cluster will be encrypted at rest or not.
+	// default=false
+	// +optional
+	Encrypted *bool `json:"encrypted,omitempty"`
+
+	// EnhancedVPCRouting specifies whether to create the cluster with enhanced VPC
+	// routing enabled. To create a cluster that uses enhanced VPC routing, the
+	// cluster must be in a VPC. For more information, see Enhanced VPC Routing
+	// (https://docs.aws.amazon.com/redshift/latest/mgmt/enhanced-vpc-routing.html)
+	// in the Amazon Redshift Cluster Management Guide.
+	// If this option is true, enhanced VPC routing is enabled.
+	// default=false
+	// +optional
+	EnhancedVPCRouting *bool `json:"enhancedVPCRouting,omitempty"`
+
+	// FinalClusterSnapshotIdentifier is the identifier of the final snapshot
+	// that is to be created immediately before deleting the cluster.
+	// If this parameter is provided, SkipFinalClusterSnapshot must be false.
+	// Constraints:
+	//    * Must be 1 to 255 alphanumeric characters.
+	//    * First character must be a letter.
+	//    * Cannot end with a hyphen or contain two consecutive hyphens.
+	// +optional
+	FinalClusterSnapshotIdentifier *string `json:"finalClusterSnapshotIdentifier,omitempty"`
+
+	// FinalClusterSnapshotRetentionPeriod is the number of days that
+	// a manual snapshot is retained.
+	// If the value is -1, the manual snapshot is retained indefinitely.
+	// The value must be either -1 or an integer between 1 and 3,653.
+	// default -1
+	// +optional
+	FinalClusterSnapshotRetentionPeriod *int64 `json:"finalClusterSnapshotRetentionPeriod,omitempty"`
+
+	// HSMClientCertificateIdentifier specifies the name of the HSM client certificate
+	// the Amazon Redshift cluster uses to retrieve the data encryption keys stored in an HSM.
+	// +optional
+	HSMClientCertificateIdentifier *string `json:"hsmClientCertificateIdentifier,omitempty"`
+
+	// HSMConfigurationIdentifier specifies the name of the HSM configuration that
+	// contains the information the Amazon Redshift cluster can use to retrieve
+	// and store keys in an HSM.
+	// +optional
+	HSMConfigurationIdentifier *string `json:"hsmConfigurationIdentifier,omitempty"`
+
+	// IAMRoles is a list of AWS Identity and Access Management (IAM) roles that can be used
+	// by the cluster to access other AWS services. You must supply the IAM roles
+	// in their Amazon Resource Name (ARN) format. You can supply up to 10 IAM roles
+	// in a single request.
+	// A cluster can have up to 10 IAM roles associated with it at any time.
+	// kubebuilder:validation:MaxItems=10
+	// +optional
+	IAMRoles []string `json:"iamRoles,omitempty"`
```

Comment:
I think you can add a reference to the IAMRole custom resource here.

Reply:
Indeed.